File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/DataStatistics2.cpp
#include <vector>
#include <numeric>
#include <algorithm>
#include <cmath>
#include <stdexcept>

class DataStatistics2 {
private:
    std::vector<double> data;

    double round_to_two(double val) {
        return std::round(val * 100.0) / 100.0;
    }

public:
    DataStatistics2(std::vector<double> data) : data(std::move(data)) {}

    double get_sum() {
        return std::accumulate(data.begin(), data.end(), 0.0);
    }

    double get_min() {
        if (data.empty()) throw std::invalid_argument("empty data");
        return *std::min_element(data.begin(), data.end());
    }

    double get_max() {
        if (data.empty()) throw std::invalid_argument("empty data");
        return *std::max_element(data.begin(), data.end());
    }

    double get_variance() {
        if (data.empty()) throw std::invalid_argument("empty data");
        double mean = get_sum() / data.size();
        double sq_sum = std::inner_product(data.begin(), data.end(), data.begin(), 0.0);
        double variance = sq_sum / data.size() - mean * mean;
        return round_to_two(variance);
    }

    double get_std_deviation() {
        if (data.empty()) throw std::invalid_argument("empty data");
        double mean = get_sum() / data.size();
        double sq_sum = std::inner_product(data.begin(), data.end(), data.begin(), 0.0);
        double variance = sq_sum / data.size() - mean * mean;
        return round_to_two(std::sqrt(variance));
    }

    double get_correlation() {
        if (data.empty()) throw std::invalid_argument("empty data");
        // For a 1D array, correlation with itself is always 1.0
        return 1.0;
    }
};

File: data/derived_cot/rq1_traces/translation/qwen/java_to_py/code_output/DataStatistics2.py
import math

class DataStatistics2:
    def __init__(self, data_list):
        self.data = [float(x) for x in data_list]
    
    def getSum(self):
        return sum(self.data) if self.data else 0.0
    
    def getMin(self):
        return min(self.data) if self.data else float('nan')
    
    def getMax(self):
        return max(self.data) if self.data else float('nan')
    
    def getVariance(self):
        n = len(self.data)
        if n == 0:
            return float('nan')
        mean = self.getSum() / n
        squared_diffs = [(x - mean) ** 2 for x in self.data]
        return sum(squared_diffs) / n
    
    def getStdDeviation(self):
        variance = self.getVariance()
        if math.isnan(variance):
            return float('nan')
        return math.sqrt(variance)
    
    def getCorrelation(self):
        return 1.0